src/SearchWrapper.cpp

```diff
@@ -0,0 +1,163 @@
+#include <Rcpp.h>
+#include <xapian.h>
+#include <cstdlib>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+using namespace std;
+
+Rcpp::List convertToDataFrame(Rcpp::List list) {
+  Rcpp::List dataFrame = clone(list);
+  Rcpp::GenericVector aRow = dataFrame(0);
+  Rcpp::StringVector rowNames(aRow.length());
+  
+  for (int i = 0; i < aRow.length(); ++i) {
+    char name[5];
+    sprintf(&(name[0]), "%d", i+1);
+    rowNames(i) = name;
+  }
+  
+  dataFrame.attr("row.names") = rowNames;
+  Rcpp::StringVector colNames(dataFrame.length());
+  
+  int ascii=97;
+  for (int j = 0; j < dataFrame.length(); ++j) {
+    char name[6];
+    //sprintf(&(name[0]), "%d", j+1);
+    sprintf(&(name[0]), "%c", char(ascii++));
+    colNames(j) = name;
+  }
+  
+  dataFrame.attr("names") = colNames;
+  dataFrame.attr("class") = "data.frame";
+  return dataFrame;
+}
+
+int columnNo( const string & data){
+  int col=0;
+  std::istringstream stream(data);
+  std::string line;
+  while(std::getline(stream, line)) {
```

Comment:
Your `for` loop has a space after the keyword, so again consistency suggests you should use that for `while` also.